// skyflow-node/lib/api/Tokens.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _axios = _interopRequireDefault(require("axios"));

var _VaultNotFoundException = _interopRequireDefault(require("../CustomErrors/VaultNotFoundException"));

var Tokens = {
  getRecordByToken: function getRecordByToken(token, options, callback) {
    var _this = this;

    return this.callApi(function (_ref) {
      var token = _ref.token,
          options = _ref.options;
      return _axios.default.get(_this.vaultUrl + '/' + 'tokens/' + token, {
        headers: _this.defaultHeaders,
        params: {
          redaction: options.redaction
        }
      }).then(function (res) {
        if (callback) {
          callback(res.data);
        }

        return res.data;
      }).catch(function (err) {
        return err && err.response && err.response.data;
      });
    }, {
      token: token,
      options: options
    });
  },
  getBulkRecordsByTokens: function getBulkRecordsByTokens(tokens, options, callback) {
    var _this2 = this;

    return this.callApi(function (_ref2) {
      var tokens = _ref2.tokens,
          options = _ref2.options;
      return _axios.default.get(_this2.vaultUrl + '/' + 'tokens', {
        params: {
          token_ids: tokens,
          redaction: options.redaction
        }
      }, {
        headers: _this2.defaultHeaders
      }).then(function (res) {
        if (callback) {
          callback(res.data);
        }

        return res.data;
      }).catch(function (err) {
        return err && err.response && err.response.data;
      });
    }, {
      tokens: tokens,
      options: options
    });
  }
};
var _default = Tokens;
exports.default = _default;
